src/svgdevicecontext.cpp

```diff
@@ -974,6 +976,33 @@ void SvgDeviceContext::AppendIdAndClass(std::string gId, std::string baseClass,
     m_currentNode.append_attribute("class") = baseClass.c_str();
 }
 
+void SvgDeviceContext::AppendAdditionalAttributes(Object *object)
+{
+    std::string baseClass = object->GetClassName();
+    std::transform(baseClass.begin(), baseClass.begin() + 1, baseClass.begin(), ::tolower);
+    for (std::string attributeLine : m_svgAdditionalAttributes) {
+        std::string className
+            = attributeLine.substr(0, attributeLine.find("@")); // parse element@attribute: "note@pname"
+        if (className == baseClass) { // if appropriate class name...
+            std::string attributeName = attributeLine.substr(attributeLine.find("@") + 1);
+            ArrayOfStrAttr attributes;
+            object->GetAttributes(&attributes);
+            ArrayOfStrAttr::iterator iter;
+            bool found = false;
+            for (iter = attributes.begin(); iter != attributes.end(); ++iter) {
+                if (attributeName == (*iter).first) // ...and attribute exists, add it to SVG element
+                {
+                    m_currentNode.append_attribute(("data-" + attributeName).c_str()) = (*iter).second.c_str();
+                    found = true;
+                }
+            }
+            if (!found)
+                LogWarning(
+                    "No attribute '%s' in '%s', nothing added to SVG.", attributeName.c_str(), className.c_str());
+        }
+    }
+}
```

Comment:
The lookup will be simpler here with the `<std::multimap>`. Something  like
```cpp
std::pair<auto first, auto second> range = m_svgAdditionalAttributes.equal_range(object->GetClassId());
for (auto it = range.first; it != range.second; ++it) {
    // it->second will be the attribute name
    // add it to the current note
}

Reply:
We still have to check whether the requested attribute exists in that element.